File: app_frontend/src/types/vendedor.ts
// src/types/vendedor.ts
import { Cliente } from './cliente';

export interface VendedorClientePorcentajeBase {
  cliente_id: number;
  porcentaje_bono: number;
}

export interface VendedorClientePorcentajeCreate extends VendedorClientePorcentajeBase {}

export interface VendedorClientePorcentajeUpdate {
    porcentaje_bono: number;
}

export interface VendedorClientePorcentaje extends VendedorClientePorcentajeBase {
  id: number;
  cliente?: Cliente;
}

export interface VendedorBase {
  nombre_completo: string;
  rut: string;
  sueldo_base: number;
}

export interface VendedorCreate extends VendedorBase {
  asignaciones?: VendedorClientePorcentajeCreate[];
}

export interface VendedorUpdate extends Partial<VendedorBase> {}

export interface Vendedor extends VendedorBase {
  id: number;
  created_at?: string | null;
  updated_at?: string | null;
  clientes_asignados: VendedorClientePorcentaje[];
}

// --- NUEVA INTERFAZ AÑADIDA ---
// Esta es la interfaz para la lista simplificada.
export interface VendedorSimple {
  id: number;
  nombre_completo: string;
}

File: app_backend/app/schemas/vendedor.py
# app/schemas/vendedor.py
from pydantic import BaseModel, constr, Field
from typing import Optional, List
from datetime import datetime
from .cliente import Cliente as ClienteSchema # Usando import relativo

# --- Schemas para VendedorClientePorcentaje ---
class VendedorClientePorcentajeBase(BaseModel):
    cliente_id: int
    porcentaje_bono: float = Field(..., gt=0, le=1)

class VendedorClientePorcentajeCreate(VendedorClientePorcentajeBase):
    pass

class VendedorClientePorcentajeUpdate(BaseModel):
    porcentaje_bono: float = Field(..., gt=0, le=1)

class VendedorClientePorcentaje(VendedorClientePorcentajeBase):
    id: int
    cliente: Optional[ClienteSchema] = None

    class Config:
        from_attributes = True

# --- Schemas para Vendedor ---
class VendedorBase(BaseModel):
    nombre_completo: str = Field(..., min_length=3, max_length=255)
    rut: constr(min_length=7, max_length=12)
    sueldo_base: float = Field(..., ge=0)

class VendedorCreate(VendedorBase):
    asignaciones: Optional[List[VendedorClientePorcentajeCreate]] = []

class VendedorUpdate(BaseModel):
    nombre_completo: Optional[str] = Field(None, min_length=3, max_length=255)
    rut: Optional[constr(min_length=7, max_length=12)] = None
    sueldo_base: Optional[float] = Field(None, ge=0)

class VendedorInDBBase(VendedorBase):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class Vendedor(VendedorInDBBase):
    clientes_asignados: List[VendedorClientePorcentaje] = []

class VendedoresResponse(BaseModel):
    items: List[Vendedor]
    total_count: int

# --- NUEVO SCHEMA AÑADIDO ---
# Este es el schema para la lista simplificada que necesita el formulario.
class VendedorSimple(BaseModel):
    id: int
    nombre_completo: str

    class Config:
        from_attributes = True
